server/src/main/java/io/helidon/extensions/mcp/server/McpServerFeature.java

```diff
@@ -521,16 +519,122 @@ private void resourcesReadRpc(JsonRpcRequest req, JsonRpcResponse res) {
                                .parameters(parameters)
                                .features(features)
                                .protocolVersion(session.protocolVersion())
+                               .context(session.context())
                                .build());
         features.progress().stopSending();
         res.result(readResource(resourceUri, contents));
         sendResponse(req, res, session, features, requestId);
     }
 
     private void resourceSubscribeRpc(JsonRpcRequest req, JsonRpcResponse res) {
+        Optional<McpSession> foundSession = findSession(req);
+        if (foundSession.isEmpty()) {
+            res.status(Status.NOT_FOUND_404).send();
+            return;
+        }
+        McpSession session = foundSession.get();
+        McpParameters parameters = new McpParameters(req.params(), req.params().asJsonObject());
+        JsonValue requestId = req.rpcId().orElseThrow(() -> new McpException("request id is required"));
+        String resourceUri = parameters.get("uri").asString().orElse(null);
+
+        // check if resource exists
+        Optional<McpResource> resource = config.resources().stream()
+                .filter(r -> resourceUri.equals(r.uri()))
```

Comment:
If `resourceUri` is `null`, this throw an NPE. Same comment for the unsubscribe method.

Reply:
Fixed